osu.Game/Screens/Play/Player.cs

```diff
@@ -285,6 +287,10 @@ private void load(AudioManager audio, OsuConfigManager config, OsuGameBase game)
             ScoreProcessor.HasCompleted.ValueChanged += updateCompletionState;
             HealthProcessor.Failed += onFail;
 
+            // Keep track of whether the storyboard ended after the playable portion
+            if (DimmableStoryboard.HasStoryboardEnded != null)
+                DimmableStoryboard.HasStoryboardEnded.ValueChanged += updateCompletionState;
```

Comment:
Wouldn't this cause `PrepareScoreForResultAsync` to be called twice? also what happens if the storyboard is less than gameplay, sounds like there should be further logic in `updateCompletionState` to consider completion when both score processor and storyboard are done.

Reply:
To me it feels like gameplay completion and storyboard completion should not be using this same code path as currently written. Both conditions need to be checked before proceeding with exit.

Reply:
Is this a better approach?

```cs
if (DimmableStoryboard.HasStoryboardEnded != null)
{
    DimmableStoryboard.HasStoryboardEnded.ValueChanged += (storyboardEnded) =>
    {
        if (storyboardEnded.NewValue && ScoreProcessor.HasCompleted.Value)
            scheduleCompletion();
    };
}
```